Add ARM64 docker build

reqlez opened the pull request:
ARM64 is widely used nowadays, so this PR covers ARM64 users:

- Modify Dockerfile to add sbt image that supports arm64
- Change docker publish GH action to build and publish arm64 image.

Personally, I run eclipse temurin jre everywhere, however, you are welcome to change it if this is not what you prefer.

pragmaxim commented:
Hi @reqlez this code is about to be removed as I changed from Cassandra to simply just a JVM process. Also it needs a stronger machine to synchronize so I believe ARM64 is not really suitable at all for an explorer, so even if I managed to keep it, it would be just a risk of switching from a well tested docker image. Sorry

reqlez commented:
> Hi @reqlez this code is about to be removed as I changed from Cassandra to simply just a JVM process. Also it needs a stronger machine to synchronize so I believe ARM64 is not really suitable at all for an explorer, so even if I managed to keep it, it would be just a risk of switching from a well tested docker image. There are more factors like Java version, etc. This is really "thin ice" area )) Sorry

My ARM64 machines are faster than a lot of AMD machines out there, are you thinking that the only people that use ARM64 are Raspberry Pi users LOL  I run production grade servers on ARM64 all the time.

Also, I switched the Ergo node repo to that image, and nobody complained ;-)

https://github.com/ergoplatform/ergo/blob/master/Dockerfile

But up to you, this is your repo, of course. If you don't feel it's necessary, don't merge, however, I will be running this explorer on ARM64 anyways :P

pragmaxim commented:
I have absolutely no time to make the old docker infrastructure work and test it so accepting your PR would most likely mean that somewhen in December I will hit some strange error originating from using different docker image and java version and will have no clue what is going on.

Let's have it open, thanks @reqlez

reqlez commented:
> I have absolutely no time to make the old docker infrastructure work and test it so accepting your PR would most likely mean that somewhen in December I will hit some strange error originating from using different docker image and java version and will have no clue what is going on.
> 
> Let's have it open, thanks @reqlez

I hope you will consider adding arm64 support in the near future, thanks you!

If you ever need any docker/infrastructure help, I have a bit of experience in that, so please don't be shy, always happy to help a fellow Ergo community member.
